GenerationExpansionPlanning/src/relaxation.jl

```diff
@@ -1,16 +1,43 @@
 export relaxation_iteration
 """
-    relaxation_iteration(data::ExperimentData, k=2)::(ExperimentData, Vector{Set{Symbol}})
-
     Create clusters and returns merged Dataframes
 """
-function relaxation_iteration(data::ExperimentData, dendogram::Hclust, k=2)::Tuple{ExperimentData, Vector{Set{Symbol}}}
+function relaxation_iteration(data::ExperimentData, k::Integer=2)::Tuple{ExperimentData, Vector{Set{Symbol}}}
     # Create k clusters
     clusters = create_clusters(data, dendogram, k)
     return (merge_within_clusters(data, clusters), clusters)
 end
 
-function create_clusters(data::ExperimentData, dendogram::Hclust, k::Integer)::Vector{Set{Symbol}}
+function create_clusters(data::ExperimentData, k::Integer, linkage::Symbol=:complete)::Vector{Set{Symbol}}
+    edges = []
+    capacities = []
+    location_indices = Dict(location => i for (i, location) ∈ enumerate(data.locations))
+
+    for line in eachrow(data.transmission_capacities)
+        source = location_indices[line.from]
+        dest = location_indices[line.to]
+        capacity = line.capacity
+        push!(edges, (source, dest))
+        push!(capacities, capacity)
+    end
+
+    # Create a distance matrix where the distance is inversely related to capacity
+    n = length(data.locations)
+    dist_matrix = fill(Inf, n, n)
+
+    for (i, edge) in enumerate(edges)
+        source, dest = edge
+        dist = 1 / capacities[i]
+        dist_matrix[source, dest] = dist
+        dist_matrix[dest, source] = dist  # Assuming undirected graph
+    end
+
+    # Perform hierarchical clustering
+    dendogram = hclust(dist_matrix, linkage=linkage, branchorder=:optimal)
+    
+    p = plot(dendogram, xticks=false)
+    savefig(p, "dendrogram.pdf")
+    
```

Comment:
This should be included now in a function in experiment, our reasoning for that was that it hclust is probably the best datastructure for keeping track of merged nodes, so it is usefull to create it only once